File: Autobase/Backend/Autobase/src/main/java/com/example/Autobase/dto/response/trip/TripResponseDto.java
package com.example.Autobase.dto.response.trip;

import lombok.Data;

import java.time.LocalDateTime;

@Data
public class TripResponseDto {
    private Long id;
    private String origin;
    private String destination;
    private String status;
    private Long driverId;
    private Long carId;
    private LocalDateTime startedAt;
    private LocalDateTime completedAt;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: Autobase/Frontend/Autobase/src/app/models/dto/response/trip/TripResponseDto.ts
import { TripStatus } from "../../../enums/TripStatus";

export interface TripResponseDto {
route: any;
dateTime: string|number|Date;
car: any;
  id: number;
  origin: string;
  destination: string;
  status: TripStatus;
  driverId: number;
  carId: number;
  startedAt?: string;           
  completedAt?: string;         
  createdAt: string;
  updatedAt: string;
}